Source/ARTAuth.m

```diff
@@ -548,30 +548,43 @@ - (void)authorize:(ARTTokenParams *)tokenParams options:(ARTAuthOptions *)authOp
     __weak id<ARTAuthDelegate> lastDelegate = self.delegate;
 
     NSString *authorizeId = [[NSUUID new] UUIDString];
+    __block BOOL hasBeenExplicitlyCanceled = false;
     // Request always a new token
     [self.logger verbose:@"RS:%p ARTAuthInternal [authorize.%@, delegate=%@]: requesting new token", _rest, authorizeId, lastDelegate ? @"YES" : @"NO"];
     NSObject<ARTCancellable> *task;
     self->_authorizationsCount += 1;
     task = [self _requestToken:currentTokenParams withOptions:replacedOptions callback:^(ARTTokenDetails *tokenDetails, NSError *error) {
         self->_authorizationsCount -= 1;
 
-        void (^successBlock)(void) = ^{
+        void (^successCallbackBlock)(void) = ^{
             [self.logger verbose:@"RS:%p ARTAuthInternal [authorize.%@]: success callback: %@", self->_rest, authorizeId, tokenDetails];
             if (callback) {
                 callback(tokenDetails, nil);
             }
         };
 
-        void (^failureBlock)(NSError *) = ^(NSError *error) {
+        void (^failureCallbackBlock)(NSError *) = ^(NSError *error) {
```

Comment:
In this case, not only should the block pointer be `const` but the `error` arg passed to the callback should be too, for strict correctness. I've not run this by the compiler but if my rusty Objective-C memories are correct then that would like this:

```suggestion
        void (^const failureCallbackBlock)(NSError *) = ^(NSError *const error) {
```

Reply:
changed here https://github.com/ably/ably-cocoa/pull/1118/commits/23b90b26f1002fe137b9013d0a7b7e577e1dc8ca

Reply:
Haven't seen that extensive usage of `const` before in objc. Basically it's mimicking swift `let` which is probably right thing to do, but definitely is not common practice. IMHO.

Reply:
I agree with @maratal, but it's not a PR merge blocker, and to be honest, I've never seen this before.